### PR TITLE
specification for adding tendrl-common utils

### PR DESCRIPTION
**Pull request by nnDarshan**
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@shtripat @nthomas-redhat @brainfunked @r0h4n Please review

**Review by shtripat — COMMENTED** [review]
LGTM

File: specs/add_utils_to_common.adoc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+adding utils to tendrl-common
+
+The expected usage of utils is as follows:
+
+[source,python]
+----
+# using command utility
+
+from tendrl.common.utils.command import Command
+command = Command("ls -la") # it takes command as argument
```
Comment (r0h4n) [inline specs/add_utils_to_common.adoc:50]: We need a white list of known commands which any module can refer to, dont allow every command to be executed

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: specs/add_utils_to_common.adoc (hunk 1 above)
Comment (nnDarshan) [reply specs/add_utils_to_common.adoc:50]: @r0h4n Good point. Will maintain a file in tendrl-common utils, to have a list of valid commands that can be executed from command util. This file will be used to validate the command passed by other components.
sounds okay ?

**Comment by nnDarshan** [comment]
Issue: https://github.com/Tendrl/specifications/issues/72

File: specs/add_utils_to_common.adoc (hunk 1 above)
Comment (r0h4n) [reply specs/add_utils_to_common.adoc:50]: If the command execution stuff is in commons then yes. Otherwise it should go in the node-agent

Comment (nnDarshan) [reply specs/add_utils_to_common.adoc:50]: yes the command-executor will be in common. Have included this info in the latest commit. Please take a look.

**Review by r0h4n — APPROVED** [review]

**Pull request merged**